blog/02-15-2023-build-feature-rich-persistent-dapp-on-celo-using-wagmi/build-feature-rich-persistent-dapp-on-celo-using-wagmi.md

```diff
@@ -0,0 +1,747 @@
+
+- The _watch_ property listens to changes in the balance and notifies to update the frontend. 
+
+- Alternatively, you can optionally set the 'token' property in place of 'address' to fetch the native token balance instead of $CELO's.
+
+> Note: Wagmi's level of persistence is higher compared to some of the other libraries.
+
+```ts
+import * as React from "react";
+import AppBar from "@mui/material/AppBar";
+import Box from "@mui/material/Box";
+import Toolbar from "@mui/material/Toolbar";
+import Typography from "@mui/material/Typography";
+import Container from "@mui/material/Container";
+import { useAccount, useBalance } from "wagmi";
+import { ConnectKitButton } from 'connectkit'
+
+function Header() {
+  const { address } = useAccount();
+  const { data } = useBalance({
+    address: address,
+    formatUnits: 'ether',
+    watch: true
+  });
+
+  return (
+    <AppBar position="fixed" sx={{ background: "#fff" }} elevation={0}>
+      <Container maxWidth="xl">
+        <Toolbar disableGutters>
+          <Typography
+            variant="h6"
+            noWrap
+            component="a"
+            href="/"
+            sx={{
+              mr: 2,
+              display: { xs: "none", md: "flex" },
+              fontFamily: "monospace",
+              fontWeight: 700,
+              letterSpacing: ".3rem",
+              color: "purple",
+              textDecoration: "none"
+            }}
+          >
+            <span style={{ fontSize: "35px" }}>Swap</span>Lab
+          </Typography>
+          <Typography
+            variant="h5"
+            noWrap
+            component="a"
+            href=""
+            sx={{
+              mr: 2,
+              display: { xs: "flex", md: "none" },
+              flexGrow: 1,
+              fontFamily: "monospace",
+              fontWeight: 700,
+              letterSpacing: ".3rem",
+              color: "purple",
+              textDecoration: "none"
+            }}
+          >
+            <span style={{ fontSize: "35px" }}>D</span>Rythm
+          </Typography>
+          <Box sx={{ flexGrow: 1, display: "flex", justifyContent: "space-around", alignItems: 'center' }}>
+            <Typography variant="body1" sx={{color: 'purple', fontWeight: 'bolder'}}>{`${data?.formatted? data.formatted : '0.00'} $Celo`}</Typography>
+            <ConnectKitButton />
+          </Box>
+        </Toolbar>
+      </Container>
+    </AppBar>
+  );
+}
+
+export default Header;
+```
+
+For example, I displayed the balance of the connected account in the header to showcase the usage of the `useAccount` and `useBalance` hooks through property destructuring. The 'data' property returned by 'useBalance' is also an object with information regarding the balance of a connected account. From the object, we accessed _data?.formatted_ - an already formatted value of _data?.value_ which is in bigNumber format. Much of the conversion task is done for us. 
```

Comment:
For example, I displayed the balance of the connected account in the header to showcase the usage of the `useAccount` and `useBalance` hooks through property destructuring. The 'data' property returned by 'useBalance' is also an object with information regarding the balance of a connected account. From the object, we accessed _data?.formatted_ - an already formatted value of _data?.value_, which is in bigNumber format. Much of the conversion task is done for us.